Stub in bcReachabilityAnalyzer off ciMethod

DanHeidinga opened the pull request:
The reachabilityAnalyzer walks the ciBytecodeStream to find the
classes that need to be loaded and the methods to process next.

It should have a similar design to the bcEscapeAnalyzer.

Add stubs to ciMethod to be able to create one.  The rest of the
code for this will be pulled across from the StaticAnalyzer in
the djh/exploration branch

Signed-off-by: Dan Heidinga <heidinga@redhat.com>

DanHeidinga committed:
Stub in bcReachabilityAnalyzer off ciMethod

The reachabilityAnalyzer walks the ciBytecodeStream to find the
classes that need to be loaded and the methods to process next.

It should have a similar design to the bcEscapeAnalyzer.

Add stubs to ciMethod to be able to create one.  The rest of the
code for this will be pulled across from the StaticAnalyzer in
the djh/exploration branch

Signed-off-by: Dan Heidinga <heidinga@redhat.com>

DanHeidinga commented:
fyi @ashu-mehra

ashu-mehra commented on src/hotspot/share/ci/ciMethod.cpp:
```diff
@@ -1278,6 +1286,18 @@ BCEscapeAnalyzer  *ciMethod::get_bcea() {
 }
 
+BCReachabilityAnalyzer  *ciMethod::get_bcra() {
+#ifdef COMPILER2 //Static Analyzer
```
I guess we don't need to protect this code with `COMPILER2`.

DanHeidinga replied:
Agreed.  I've commented out the #else branch as we'll want effectively the same code using our own static analyzer define in the future

ashu-mehra commented:
This looks fine except that minor change.

DanHeidinga committed:
Remove #ifdef for COMPILER2

I've left the code commented out as we'll want to bring it back when
we have a proper ifdef for the static analyzer

Signed-off-by: Dan Heidinga <heidinga@redhat.com>
